src/documentation/programming-guide.md

```diff
@@ -1460,9 +1460,16 @@ An example might be if your pipeline reads log records from an input file, and e
 
 > **NOTE:** This content applies only to the Beam SDK for Java. The Beam SDK for Python does not support triggers.
 
-When collecting and grouping data into windows, Beam must determine when to emit the aggregated results of each window. If you use Beam's default windowing configuration and [default trigger](#default-trigger), Beam outputs the aggregated result when it [estimates all data has arrived](#watermarks-late-data), and discards all subsequent data for that window.
+When collecting and grouping data into windows, Beam uses **triggers** to determine when to emit the aggregated results of each window (referred to as a *pane*). If you use Beam's default windowing configuration and [default trigger](#default-trigger), Beam outputs the aggregated result when it [estimates all data has arrived](#watermarks-late-data), and discards all subsequent data for that window.
 
-**Triggers** allow you to change this default behavior and specify when Beam emits the aggregated results for each window (referred to as a *pane*). At a high level, triggers provide two additional capabilities compared to simply outputting at the end of a window:
+You can set triggers for your `PCollection`s to change this default behavior and specify when to emit the aggregated results for each window. Beam provides a number of pre-built triggers that you can set:
+
+*   **Event time triggers**. These triggers operate on the event time, as indicated by the timestamp on each data element. Beam's default trigger is event time-based.
+*   **Processing time triggers**. These triggers operate on the processing time -- the time when the data element is processed at any given stage in the pipeline.
+*   **Data-driven triggers**. These triggers operate by examining the data as it arrives in each window, and firing when that window has received a certain number of data elements.
+*   **Composite triggers**. These triggers combine multiple triggers in some logical way.
+
+At a high level, triggers provide two additional capabilities compared to simply outputting at the end of a window:
 
 *   Triggers allow Beam to emit early results, before all the data in a given window has arrived.
 *   Triggers allow processing of late data.
```

Comment:
by triggering after the end of the event time watermark has past the end of the window